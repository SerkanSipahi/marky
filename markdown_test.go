package marky_test

import (
	"github.com/serkansipahi/marky"
	"io/ioutil"
	"testing"
)

func TestNewMarkdown(t *testing.T) {

	markdownTemplate, _ := ioutil.ReadFile("./markdown_test.md")
	expectedHeaders, _ := ioutil.ReadFile("./markdown_test_expected.txt")

	_, markdown := marky.NewMarkdown(string(markdownTemplate))
	code := markdown.Compile()

	if code != string(expectedHeaders) {
		t.Error(
			"expected", string(expectedHeaders),
			"got", code,
		)
	}

}
