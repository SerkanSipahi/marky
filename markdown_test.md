#Description
### Hello my friend
Hi, whats going on
##Nothing everyting is fine
[Link1](http://www.example-1.com) - [Link2](http://www.example-2.com)
## Here is an [Link1](http://www.example-3.com) in header
Thats an p with a [Link2](http://www.example-4.com) link
Waw really ***awesome3*** -- ***awesome3***!
Waw really ***awesome3*** -- ***awesome3*** [Link5](http://www.example-5.com)!
## Waw really ***awesome3*** -- [Link5](http://www.example-6.com) -- ***awesome6***!
Waw really **awesome2** -- ***awesome3*** -- *awesome2* -- [Link5](http://www.example-6.com)!
##### Waw really *awesome1* -- *awesome2*!