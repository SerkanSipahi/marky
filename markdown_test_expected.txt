<h1>Description</h1>
<h3>Hello my friend</h3>
<p>Hi, whats going on</p>
<h2>Nothing everyting is fine</h2>
<p><a href='http://www.example-1.com'>Link1</a> - <a href='http://www.example-2.com'>Link2</a></p>
<h2>Here is an <a href='http://www.example-3.com'>Link1</a> in header</h2>
<p>Thats an p with a <a href='http://www.example-4.com'>Link2</a> link</p>
<p>Waw really <strong><em>awesome3</em></strong> -- <strong><em>awesome3</em></strong>!</p>
<p>Waw really <strong><em>awesome3</em></strong> -- <strong><em>awesome3</em></strong> <a href='http://www.example-5.com'>Link5</a>!</p>
<h2>Waw really <strong><em>awesome3</em></strong> -- <a href='http://www.example-6.com'>Link5</a> -- <strong><em>awesome6</em></strong>!</h2>
<p>Waw really <strong>awesome2</strong> -- <strong><em>awesome3</em></strong> -- <em>awesome2</em> -- <a href='http://www.example-6.com'>Link5</a>!</p>
<h5>Waw really <em>awesome1</em> -- <em>awesome2</em>!</h5>
